etc/profile-m-z/neochat.profile

```diff
@@ -0,0 +1,75 @@
+# Firejail profile for neochat
+# Description: Matrix Client
+# This file is overwritten after every install/update
+# Persistent local customizations
+include neochat.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.cache/KDE/neochat
+noblacklist ${HOME}/.config/KDE/neochat
+noblacklist ${HOME}/.config/neochatrc
+noblacklist ${HOME}/.config/neochat.notifyrc
+noblacklist ${HOME}/.local/share/KDE/neochat
+noblacklist ${DOWNLOADS}
+
+include disable-common.inc
+include disable-devel.inc
+include disable-exec.inc
+include disable-interpreters.inc
+include disable-passwdmgr.inc
+include disable-programs.inc
+include disable-shell.inc
+include disable-write-mnt.inc
+include disable-xdg.inc
+
+mkdir ${HOME}/.cache/KDE/neochat
+mkdir ${HOME}/.config/KDE/neochat.conf
+mkdir ${HOME}/.local/share/KDE/neochat
+mkfile ${HOME}/.config/neochatrc
+mkfile ${HOME}/.config/neochat.notifyrc
+
+whitelist ${HOME}/.cache/KDE/neochat
+whitelist ${HOME}/.config/KDE/neochat.conf
+whitelist ${HOME}/.config/neochatrc
```

Comment:
Can neochat save changed settings?

Reply:
Actually, no. I thought I'd checked that. :-( 
It works if I don't whitelist anything, but these are all the files which I can see created when I run it with `firejail --private`.

Reply:
It works if I whitelist `~/.config`. Not sure how to find out which file/subfolder I'm missing.

Reply:
I asked because that's an known issue with KDE programs (#1793).

I would say comment the `whitelist`ing (not wruc,wusc,wcv) and add a comment like "Makes settings immutable, add to your neochat.local after you configured neochat."

Reply:
Would it make sense to whitelist `~/.config`? At least that way, the rest of the user's home directory will be protected.

Reply:
Can you test if this works.
```
noblacklist ${HOME}/.config/KDE
noblacklist ${HOME}/.config/KDE/neochat
noblacklist ${HOME}/.config/neochatrc
noblacklist ${HOME}/.config/neochat.notifyrc
blacklist ${HOME}/.config/*
whitelist ${HOME}/.config
whitelist ${HOME}/.local/share/KDE/neochat
whitelist ${HOME}/.cache/KDE/neochat
include whitelist-common.inc
include disable-common.inc
include disable-programs.inc
read-only ${HOME}/.config/*
read-write  ${HOME}/.config/neochatrc
read-write  ${HOME}/.config/KDE/neochat
read-write  ${HOME}/.config/neochat.notifyrc
```

Reply:
It doesn't even remember the login if I `blacklist ${HOME}/.config/*`. If I remove just that command, it works, but still doesn't remember any changed settings.

Reply:
If I add this but comment out `read-only ${HOME}/.config/*`, I can change settings.

Reply:
So if I do
```
noblacklist ${HOME}/.cache/KDE/neochat
noblacklist ${HOME}/.config/KDE
noblacklist ${HOME}/.config/KDE/neochat
noblacklist ${HOME}/.config/neochatrc
noblacklist ${HOME}/.config/neochat.notifyrc
noblacklist ${HOME}/.local/share/KDE/neochat

blacklist ${HOME}/.config/*

mkdir ${HOME}/.config/neochat
mkdir ${HOME}/.cache/KDE/neochat
mkdir ${HOME}/.config/KDE/neochat.conf
mkdir ${HOME}/.local/share/KDE/neochat
mkfile ${HOME}/.config/neochatrc
mkfile ${HOME}/.config/neochat.notifyrc

whitelist ${HOME}/.config
whitelist ${HOME}/.cache/KDE/neochat
whitelist ${HOME}/.local/share/KDE/neochat
```
I can change settings in the program. The program can see other files in `.config`, but can't open them. However, it can access all files in `.config/KDE/`. Should I go ahead and do this?

Reply:
> mkdir ${HOME}/.config/…

Can be skipped

> all files in .config/KDE/

We can `blacklist .config/KDE/*` too

>  Should I go ahead and do this?

1. We will likely need to noblacklist some of them `grep '\.config' /etc/firejail/whitelist-common.inc`
2. @glitsj16 what do you think?

Reply:
I'm not at all familiar with KDE I'm afraid. Currently there's no mention of .config/KDE in any of our files AFAICT. @Kishore96in What do you have under ~/.config/KDE on your system? If this is a stupid question, disregard. Like I said, I never touched KDE and I'm not even sure I get the question :-)

Reply:
The main questions is: Should we this #1793 workaround?
```
noblacklist ${HOME}/.config/a
noblacklist ${HOME}/.config/b
noblacklist ${HOME}/.config/c
blacklist ${HOME}/.config/*
whitelist ${HOME}/.config
```

Pros:
- whitelisting enabled by default

Cons:
- ugly (we should add a comment that explains what we are doing)
- maintenance to be in sync with whitelist-common

Reply:
@rusty-snake Thanks for providing context.
> maintenance to be in sync with whitelist-common
Personally I would go for the strongest hardening. So bring in the #1793 workaround. We already have quite a complex maintenance situation, one more shouldn't hurt that much as long as we're aware of it (like with another comment).

Reply:
> Should I go ahead and do this?

@Kishore96in yes but add `noblacklist`s for `grep '\.config' /etc/firejail/whitelist-common.inc` (the kde/qt stuff is enough IMHO) and a comment what we are doing here (above the `blacklist ~/.config/*`?).

Reply:
> @Kishore96in yes but add `noblacklist`s for `grep '\.config' /etc/firejail/whitelist-common.inc` (the kde/qt stuff is enough IMHO) and a comment what we are doing here (above the `blacklist ~/.config/*`?).

Would it be better to do the blacklisting in a separate file (say `blacklist-common-kde.inc`) and include it here, so that this fix can be easily applied to other profiles if necessary?

Reply:
A `whitelist-1793-workaround.inc` (or kde-workaround) or what evername (I would use a `whitelist-` prefix as it is used to enable whitelisting. If not `allow-` prefix) like
```
# HEADER
# COMMENT
noblacklist ${HOME}/.config/Trolltech.conf
noblacklist ${HOME}/.config/QtProject.conf
noblacklist …
blacklist ${HOME}/.config/*
whitelist ${HOME}/.config
```
? Sound good to me.

Reply:
> blacklist .config/KDE/*

Doing this seems to make `.config/KDE/neochat` inaccessible as well (even with the explicit noblacklist earlier), so I have not done it. As of the current version of the PR, the files in `.config/KDE` are accessible to the program (this folder contains config files of some other KDE apps as well), but the program cannot access other unnecessary folders or files in .config.

Otherwise, the program works now, and settings etc. can be changed and persist.